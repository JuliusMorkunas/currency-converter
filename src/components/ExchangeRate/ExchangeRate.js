import React from 'react';
import PropTypes from 'prop-types';
import s from './ExchangeRate.module.scss';

const ExchangeRate = ({ rate, currencyFrom, currencyTo }) => {
  return (
    <div className={s.Container}>
      <span className={s.Decoration} />
      <span className={s.ExchangeRate}>
        1 {currencyFrom} = {!rate ? '...' : rate} {currencyTo}
      </span>
    </div>
  );
};

ExchangeRate.propTypes = {
  rate: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  currencyFrom: PropTypes.string,
  currencyTo: PropTypes.string,
};

export default ExchangeRate;
