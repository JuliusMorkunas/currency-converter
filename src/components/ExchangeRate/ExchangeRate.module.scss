@import '../../variables';

.Container {
  display: flex;
  align-items: center;
  margin: 20px 0;
}

.ExchangeRate {
  font-size: 18px;
  font-weight: 600;
  text-transform: uppercase;
  color: $c-blackish;
}

.Decoration {
  display: block;
  width: 11px;
  height: 11px;
  margin-right: 8px;
  border: 3px solid $c-yellow;
  border-radius: 5px;
}
