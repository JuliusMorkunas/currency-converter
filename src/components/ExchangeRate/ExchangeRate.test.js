import React from 'react';
import { render } from '@testing-library/react';
import ExchangeRate from './ExchangeRate';

it('ExchangeRate should match snapshot', () => {
  const { asFragment } = render(<ExchangeRate />);
  expect(asFragment()).toMatchSnapshot();
});
