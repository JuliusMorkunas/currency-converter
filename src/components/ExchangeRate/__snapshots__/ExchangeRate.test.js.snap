// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRate should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container"
  >
    <span
      class="Decoration"
    />
    <span
      class="ExchangeRate"
    >
      1  = ... 
    </span>
  </div>
</DocumentFragment>
`;
